token_client.go

```diff
@@ -4,18 +4,18 @@ import (
 	"encoding/json"
 	"errors"
 	"net/http"
+	"strconv"
 	"time"
 )
 
-type clientCredentialsTokenManager struct {
+type TokenClient struct {
 	ClientID      string
 	ClientSecret  string
 	NetClient     *http.Client
-	Token         *accessToken
 	TokenEndpoint string
 }
 
-type clientCredentialsManagerConfig struct {
+type TokenClientConfig struct {
```

Comment:
Now that we don't share the HTTP client, it could be created in `newTokenClient` instead of passed via config. If we leave as it is, the library's client will be forced to create a HTTP client while it doesn't need to do the same while creating the API client.